src/inference/InterfaceExtensions.jl

```diff
@@ -0,0 +1,334 @@
+module InterfaceExtensions
+
+using ..Turing
+using ..Interface
+import MCMCChains: Chains
+
+import ..Interface: AbstractTransition, sample, step!, sample_init!,
+    transitions_init, sample_end!, AbstractSampler, transition_type
+import ..Turing: Model, Sampler
+import ..RandomVariables: islinked, invlink!, getlogp, tonamedtuple
+import ..Inference: InferenceAlgorithm, ParticleInference
+
+export sample,
+       sample_init!,
+       sample_end!,
+       set_resume!,
+       resume,
+       save,
+       transition_type,
+       transitions_init,
+       transition,
+       Transition,
+       ParticleTransition,
+       Chains
+
+# Internal variable names for MCMCChains.
+const TURING_INTERNAL_VARS =
+    Dict(:internals => ["elapsed", "eval_num", "lf_eps", "lp", "weight", "le",
```

Comment:
"tree_depth" and "hamiltonian_energy" are very specific for HMC / NUTS. Would it maybe be better to include only rather generic internals here and let the respective inference algo push new internal fields if necessary? This would be more flexible.

Reply:
You could, but I don't really know how much difference it would make. It might even be worse. Turing knows all the parameters that are going to go into the `:internals` field, so we should just make a blanket statement instead of replicating dictionaries multiple times. It costs us nothing to have one dictionary that covers everything easily. 

It also means that anytime we change anything in any sampler, or even system-wide, we only ever have to go to one place to make sure it goes into the internals field. It's also incredibly easy to diagnose if something is not going into the internals field.

Reply:
Sure, I was just pointing out that the list will likely be incomplete. But I’m fine with it for now.